### ISSUE TITLE
Good practices when deploying a jsonrpsee server to production

### ISSUE THREAD

**Issue by Eikix** [paritytech/jsonrpsee#1023]
Hey guys! It's me again! I hope you're doing well. I'd like to deploy an MVP of the RPC server [we implemented on Kakarot](https://github.com/sayajin-labs/kakarot-rpc). 

I'm looking at several PaaS websites: [fly.io](fly.io), [heroku](https://www.heroku.com/), and others.

I'd like to know if there is some kind of good practice/guide you have when deploying such a backend app. I'd like to point out I've never done it before. I have experience deploying serverless apps on AWS (Infrastructure as Code and Serverless). I also have experience deploying frontend websites (using NextJS or other frameworks, SvelteKit + Netlify for instance).

TL;DR: How would you suggest one deploy [their jsonrpsee app](https://github.com/sayajin-labs/kakarot-rpc)?

**Comment by Eikix** [comment]
I guess the most generic example could be Dockerfile example for jsonrpsee?

**Comment by Eikix** [comment]
I got this so far:

```docker
FROM rust:1.66 as builder

# Set working directory
WORKDIR /usr/src/rpc

# Copy source code
COPY . .

# Build the application
RUN cargo build --all --release

# Create a new container from scratch to reduce image size
FROM debian:buster-slim

# Install any necessary dependencies
RUN apt-get update && apt-get install -y libssl-dev && rm -rf /var/lib/apt/lists/*

# Set the working directory
WORKDIR /usr/src/app

# Copy the built binary from the previous container
COPY --from=builder /usr/src/rpc/target/release/kakarot-rpc .

ENV STARKNET_RPC_URL=https://starknet-goerli.infura.io/v3/<MY_INFURA_PROJECT_ID>
ENV RUST_LOG=debug

# Expose the port that the RPC server will run on
EXPOSE 3030

# Run the binary
CMD ["./kakarot-rpc"]


```

**Comment by niklasad1** [comment]
Sorry, we that maintain this repo are no experts on deployment/devops stuff and can't really provide any recommendations how to deploy your rpc service.

**Comment by Eikix** [comment]
Okay, thanks a lot for your time:)! Have a good one
